Shared/Notifications/UserNotificationManager+Extensions.swift

```diff
@@ -11,14 +11,21 @@ import UserNotifications
 
 extension UserNotificationManager: UNUserNotificationCenterDelegate {
 
-    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification) async -> UNNotificationPresentationOptions {
+    func userNotificationCenter(_ center: UNUserNotificationCenter,
+                                willPresent notification: UNNotification) async -> UNNotificationPresentationOptions {
+
         guard notification.request.content.categoryIdentifier != "message.new" else { return [] }
+
         return [.banner, .list, .sound, .badge]
     }
 
-    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse) async {
-        
-        if let action = UserNotificationAction.init(rawValue: response.actionIdentifier) {
+    // NOTE: This delegate function seems to get called off the main thread some times which causes a
+    // crash when bringing the app out of the background.
+    @MainActor
```

Comment:
For some reason this function was getting called off the main thread which caused a crash.

Reply:
wtf.